SqlServer/Import-BacpacToSqlServer/Import-BacpacToSqlServer.psm1

```diff
@@ -160,35 +212,53 @@ function Import-BacpacToSqlServer
 
 
 
-        # And now we get to actually importing the database after constructing a connection string with validated details.
-        $ConnectionString = "Data Source=$DataSource;Initial Catalog=$DatabaseName;Integrated Security=True;"
-        & "$sqlPackageExecutablePath" /Action:Import /SourceFile:"$BacpacPath" /TargetConnectionString:"$ConnectionString"
+        $parameters = @(
+            '/Action:Import',
+            "/TargetServerName:`"$DataSource`"",
+            "/TargetDatabaseName:`"$DatabaseName`"",
+            '/TargetTrustServerCertificate:True',
+            '/TargetTimeout:0',
+            "/SourceFile:`"$BacpacPath`""
+        )
+
+        if (![string]::IsNullOrEmpty($Username) -and ![string]::IsNullOrEmpty($Password))
+        {
+            $parameters.Add("/TargetUser:`"$Username`"")
+            $parameters.Add("/TargetPassword:`"$Password`"")
+        }
```

Comment:
I get an exception here:
> Import-BacpacToSqlServer : Exception calling "Add" with "1" argument(s): "Collection was of a fixed size."

I've added `echo $parameters.GetType().FullName` before the condition and it printed `System.Object[]` so that makes sense. This should be a list. (I've temporarily moved these two lines into the `$parameters` to test the rest.)